src/udiskslinuxpartition.c

```diff
@@ -812,13 +847,29 @@ handle_resize (UDisksPartition       *partition,
       goto out;
     }
 
+  /* Wait for partition to reappear before returning, so that clients
+   * cat at least issue udisks_client_settle() afterwards to wait for
+   * the other interfaces to be present. If the resizing/fitting results
+   * in the same partition size this waiter will hit its timeout
+   * because there is no way to distinguish the reappeared partition
+   * from the original one.
+   */
```

Comment:
I would need to change libblockdev then to not write the partition table if the size does not change. Then this could work!